docs/best-practices/index.md

```diff
@@ -0,0 +1,438 @@
+
+It would be better practice therefore to use a map of objects, where there is a static key:
+```terraform
+variable "populations" {
+  type = map(object({
+    name        = string
+    description = optional(string)
+  }))
+
+  default = {
+    "first_population" = {
+      name        = "My awesome population"
+      description = "My awesome population for awesome people"
+    },
+    "second_population" = {
+      name = "My awesome second population"
+    }
+  }
+}
+
+resource "pingone_population" "my_awesome_population_map_of_objects" {
+  environment_id = pingone_environment.my_environment.id
+
+  for_each = var.populations
+
+  name        = each.value.name
+  description = each.value.description
+}
+```
+
+The above results in creation of two unique resources:
+
+* `pingone_population.my_awesome_population_list_of_objects["first_population"]`
+* `pingone_population.my_awesome_population_list_of_objects["second_population"]`
+
+In this case, if the `name` or `description` of any population changes, Terraform will correctly update the impacted resource, rather than potentially forcing a re-creation.
+
+Notice also, that if the order of the key/object pairs changes in the map, Terraform correctly calculates that there are no changes to the data with the objects themselves, because the relation of object to map key hasn't changed.  This is advantage of using `for_each` over `count`, where changing the order of items does impact the plan that Terraform calculates, because the counted index related to the data has changed.
+
+### Write and Publish Re-usable Modules
+
+When writing Terraform HCL, there are often times when collections of resources and data sources are commonly used together, or used frequently with the same, or very similar structure.  These collections of resources and data sources can be grouped together into a Terraform module.  Writing and publishing Terraform modules embodies a best practice within infrastructure as code (IaC) paradigms for several compelling reasons.
+
+Firstly, modules encapsulate and abstract complex sets of resources and configurations, promoting reusability and reducing redundancy across your infrastructure setups. This modular approach enables teams to define standardized and vetted building blocks, ensuring consistency, compliance, and reliability across deployments. Moreover, publishing these modules, either internally within an organization or publicly in the Terraform Registry, fosters collaboration and knowledge sharing. It allows others to benefit from proven infrastructure patterns, contribute improvements, and stay aligned with the latest best practices. This culture of sharing and collaboration not only accelerates development cycles but also elevates the quality of infrastructure provisioning by leveraging the collective expertise and experience of the Terraform community.
```

Comment:
First, modules 

Strike the last sentence altogether.  (This culture of sharing...)